src/npm/components/VisualiserWrapper.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import DataWrapper from '../../components/molecules/DataWrapper';
+import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client';
+import { ChartEntry } from '../../types/visualisation';
+import dataSourceMappings, { DataSourceID } from '../../utils/dataSourceMappings';
+
+const client = new ApolloClient({
+    uri: process.env.AZURE_GRAPHQL_URI,
+    cache: new InMemoryCache({}),
+});
+
+type VisualiserWrapperProps = {
+    children: (data: ChartEntry[]) => JSX.Element;
+    dataSource: DataSourceID;
+};
+
+const VisualiserWrapper: React.FC<VisualiserWrapperProps> = ({ children, dataSource }) => {
+    return (
+        <ApolloProvider client={client}>
+            <DataWrapper
+                mappingFunction={dataSourceMappings[dataSource].mapping}
+                query={dataSourceMappings[dataSource].query}
+            >
+                {(data) => children(data)}
+            </DataWrapper>
+        </ApolloProvider>
+    );
+};
```

Comment:
I'd vote for separating these two. 

The data wrapper could be part of the individual visualisations and perhaps use [higher order functions](https://anil-pace.medium.com/hof-and-hoc-in-react-4de1e27e5a4e).

The provider should be a standalone component as you should _only have one_ provider per provided object.

Reply:
So having the DataWrapper around

Reply:
Ok sure, maybe I could make the ApolloProvider a wrapper that a user should wrap around their App component? Just like a normal provider should be used